grails-app/controllers/org/pih/warehouse/reporting/ConsumptionController.groovy

```diff
@@ -509,8 +512,21 @@ class ShowConsumptionCommand implements Validateable {
         parametersHash(nullable: true)
     }
 
-    def getAvailableProperties() {
-        Holders.grailsApplication.mappingContext.getPersistentEntity(Product.class.name).persistentPropertyNames
+    static String getTypePropertyName(Class type) {
+        String shortTypeName = ClassUtils.getShortClassName(type)
+        return shortTypeName.substring(0, 1).toLowerCase(Locale.ENGLISH) + shortTypeName.substring(1)
+    }
+
+    static List<ConsumptionReportProductProperty> getAvailableProperties() {
+        List<PersistentProperty> properties =
+            Holders.grailsApplication.mappingContext.getPersistentEntity(Product.class.name).persistentProperties
+        return properties.collect { PersistentProperty it ->
+            new ConsumptionReportProductProperty([
```

Comment:
The reason I've created a class for that one is only for "IDE suggestion purposes". If I returned a generic Map here, I'd have to return `List<Map>`, so that the compiler would not know what properties I might have here, so I wouldn't have suggestions in the `_filters.gsp` about `name`, `naturalName` etc.
Let me know if I should change it to the generic Map, or I can proceed with the class if figured out a better name,

Reply:
I feel like we have a ton of other places with this "issue". And trying to address it in some places and leaving it in others will cause some chaos. Let's use a generic map for now, and we can discuss this topic on the TechHuddle as a tech debt idea.

Reply:
(unless you already have something better in mind than `ConsumptionReportProductProperty` and `DomainClassProperty` that can be used as a util class, then you can use it)

Reply:
Went with generic Map for now. Will create a tech debt ticket to handle those.